integration_test/sign_in_screen_test.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:survey_flutter_ic/navigation/route.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_screen.dart';
+import 'package:survey_flutter_ic/ui/signin/sign_in_widget_id.dart';
+
+import 'fake_data/fake_data.dart';
+import 'utils/test_util.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  signInTest();
+}
+
+void signInTest() {
+  group('SignIn Page', () {
+    late Finder emailField;
+    late Finder passwordField;
+    late Finder signInButton;
+
+    setUpAll(() async {
+      await TestUtil.setupTestEnvironment();
+    });
+
+    setUp(() {
+      emailField = find.byKey(SignInWidgetId.emailInputField);
+      passwordField = find.byKey(SignInWidgetId.passwordInputField);
+      signInButton = find.byKey(SignInWidgetId.signInButton);
+    });
+
+    testWidgets(
+        "When the sign in screen shown, it displays the Sign In screen correctly",
+        (WidgetTester tester) async {
+      await tester
+          .pumpWidget(TestUtil.pumpWidgetWithShellApp(const SignInScreen()));
+      await tester.pumpAndSettle();
+
+      expect(emailField, findsOneWidget);
+      expect(passwordField, findsOneWidget);
+      expect(signInButton, findsOneWidget);
+    });
+
+    testWidgets(
+        "When sign in with valid email and password, it navigate to Home screen",
+        (WidgetTester tester) async {
+      await FakeData.initDefault();
+      await tester
+          .pumpWidget(TestUtil.pumpWidgetWithRoutePath(routePathSignInScreen));
+      await tester.pumpAndSettle();
+      await tester.enterText(emailField, 'valid@example.com');
+      await tester.enterText(passwordField, '1111111');
+      await tester.tap(signInButton);
+      await tester.pump(const Duration(milliseconds: 200));
```

Comment:
Can we call `pumpAndSettle()` instead?
```suggestion
     await tester.pumpAndSettle();
```

Reply:
@Wadeewee In here, the `pump` is supposed to refresh the frame after the `tester.tap(signInButton)`, so there is just a single widget that need to be refreshed so `pump` should be used. And there is also `tester.pumpAndSettle()` to wait for all widgets on frame so I think we don't need to update 🙏